server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -514,16 +527,18 @@ public Query rangeQuery(
                 float l = Float.NEGATIVE_INFINITY;
                 float u = Float.POSITIVE_INFINITY;
                 if (lowerTerm != null) {
-                    l = parse(lowerTerm, false);
-                    if (includeLower == false) {
-                        l = FloatPoint.nextUp(l);
+                    l = parseToFloat(lowerTerm);
+                    if (includeLower) {
+                        l = FloatPoint.nextDown(l);
```

Comment:
I don't understand why you do `nextDown` at all here. Shouldn't the parsing itself be enough?

Reply:
Ah, yeah, those seem like sane optimizations and I am happy to wait for them.

Reply:
> if the interval is [-Infinity, +Infinity] we can return a FieldExistsQuery

This looks like it is not needed since it's already taken care of at the Lucene level.

> if the interval is [-Infinity, -Infinity] or [+Infinity, +Infinity] to return a MatchNoDocsQuery

This can be further generalised so that we return a `MatchNoDocsQuery` when `upper > lower` - I think we can make this change in Lucene rather than adding the check here: see https://github.com/apache/lucene/pull/13425